pkg/apis/core/types_namespacedcloudprofile.go

```diff
@@ -0,0 +1,82 @@
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NamespacedCloudProfile represents certain properties about a provider environment.
+type NamespacedCloudProfile struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Spec defines the provider environment properties.
+	Spec NamespacedCloudProfileSpec
+	// Most recently observed status of the NamespacedCloudProfile.
+	Status NamespacedCloudProfileStatus
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NamespacedCloudProfileList is a collection of NamespacedCloudProfiles.
+type NamespacedCloudProfileList struct {
+	metav1.TypeMeta
+	// Standard list object metadata.
+	metav1.ListMeta
+	// Items is the list of CloudProfiles.
+	Items []NamespacedCloudProfile
+}
+
+// NamespacedCloudProfileSpec is the specification of a NamespacedCloudProfile.
+type NamespacedCloudProfileSpec struct {
+	// CABundle is a certificate bundle which will be installed onto every host machine of shoot cluster targeting this profile.
+	CABundle *string
+	// Kubernetes contains constraints regarding allowed values of the 'kubernetes' block in the Shoot specification.
+	Kubernetes *KubernetesSettings
+	// MachineImages contains constraints regarding allowed values for machine images in the Shoot specification.
+	MachineImages []MachineImage
+	// MachineTypes contains constraints regarding allowed values for machine types in the 'workers' block in the Shoot specification.
+	MachineTypes []MachineType
+	// Regions contains constraints regarding allowed values for regions and zones.
+	Regions []Region
+	// SeedSelector contains an optional list of labels on `Seed` resources that marks those seeds whose shoots may use this provider profile.
+	// An empty list means that all seeds of the same provider type are supported.
+	// This is useful for environments that are of the same type (like openstack) but may have different "instances"/landscapes.
+	// Optionally a list of possible providers can be added to enable cross-provider scheduling. By default, the provider
+	// type of the seed must match the shoot's provider.
+	SeedSelector *SeedSelector
+	// VolumeTypes contains constraints regarding allowed values for volume types in the 'workers' block in the Shoot specification.
+	VolumeTypes []VolumeType
+	// Parent contains a reference to a CloudProfile it inherits from.
+	Parent CloudProfileReference
+}
+
+// NamespacedCloudProfileStatus holds the most recently observed status of the project.
+type NamespacedCloudProfileStatus struct {
+	// CloudProfile is the most recently generated CloudProfile of the NamespacedCloudProfile.
+	CloudProfileSpec CloudProfileSpec
+}
+
+// CloudProfileReference holds the information about the parent of the NamespacedCloudProfile.
+type CloudProfileReference struct {
+	// Kind contains a CloudProfile kind, must be "CloudProfile" or "NamespacedCloudProfile"
```

Comment:
According to [GEP](https://github.com/gardener/gardener/blob/master/docs/proposals/25-namespaced-cloud-profiles.md#multi-level-inheritance) , `multi-level-inheritance` was not supposed to be implemented, is the plan changed?

Reply:
No, but this is a reusable structure for storing a `CloudProfile` reference that is also used by the `Shoot` API for example. The `NamespacedCloudProfile`s static validation only allows for a `CloudProfile` kind to be the parent.

Reply:
Shouldn't the doc string also say the same, otherwise, it is confusing that by doc string one will get the sense that both `NamespacedCloudProfiles` and `CloudProfile` a valid values, which is not the case, we can update it in the future when `NamespacedCloudProfiles` is also supported?

Reply:
I adjusted the docstrings in https://github.com/gardener/gardener/pull/9440/commits/b9e1b4ed0c768287ed2cee6c2ce2c39e11b88d65 so that they don't mention which `CloudProfile`s can be used, to keep the struct as general as possible.